specification/indices/_types/DataStream.ts

```diff
@@ -47,6 +53,11 @@ export class DataStream {
    *  If `true`, the data stream allows custom routing on write request.
    */
   allow_custom_routing?: boolean
+  /**
+   * Information about failure store backing indices
+   *
+   */
+  failure_store?: FailureStore
```

Comment:
I've made it optional because it's behind a feature flag (and undocumented): https://github.com/elastic/elasticsearch/blob/03384b5f6ffdbafbb57c261c798911303801c4d1/server/src/main/java/org/elasticsearch/action/datastreams/GetDataStreamAction.java#L350-L360. There's a `feature_flag` annotation in the specification, but only for requests.

When the feature flag go away, this will no longer be optional.